### PR TITLE
update to version 2.0.21

### PR DESCRIPTION
**Pull request by M-Waszkiewicz-Anaconda**

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
   version: {{ version }}
 
 source:
-  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
-  sha256: faa85e053c0b1be4d5585b0858d3a511d2cd10201802e8676060fd0a109e5869
+  url: https://github.com/unbit/{{ name }}/archive/{{ version }}.tar.gz
+  sha256: 35a30d83791329429bc04fe44183ce4ab512fcf6968070a7bfba42fc5a0552a9
 
 build:
   number: 3
```
Comment (skupr-anaconda) [inline recipe/meta.yaml]: The build number must be 0 for a new version

**Review by skupr-anaconda — COMMENTED** [review]
Also, missing `pip check` in `test/commands,` and `pip` in `test/requires`.
Missing dev_url and doc_url

**Comment by akabanovs** [comment]
Nothing much to add to @skupr-anaconda review. One thing I wanted to check:

Shall we pin icu and openssl in the host section to use whatever is in aggregate cbc? 

 - icu  = {{ icu }}
 - openssl  ={{ openssl }}
 
For this case these pinnings will not change anything as the ones in the cbc are also the latest versions we have on the main channel, but might be just good for us to know these versions come from global cbc (defined there) and these versions are compatible.

**Comment by akabanovs** [comment]
Also, I think it is better to use this url to download the tar file from:

https://files.pythonhosted.org/packages/b3/8e/b4fb9f793745afd6afcc0d2443d5626132e5d3540de98f28a8b8f5c753f9/uwsgi-2.0.21.tar.gz

This url is used here:
https://pypi.org/project/uWSGI/#files

and also on the uwsgi official site:
https://uwsgi-docs.readthedocs.io/en/latest/Changelog-2.0.21.html

**Comment by M-Waszkiewicz-Anaconda** [comment]
> Also, I think it is better to use this url to download the tar file from:
> 
> https://files.pythonhosted.org/packages/b3/8e/b4fb9f793745afd6afcc0d2443d5626132e5d3540de98f28a8b8f5c753f9/uwsgi-2.0.21.tar.gz
> 
> This url is used here: https://pypi.org/project/uWSGI/#files
> 
> and also on the uwsgi official site: https://uwsgi-docs.readthedocs.io/en/latest/Changelog-2.0.21.html

I asked yesterday about it on our team channel, the precendence of urls should be:
1. pypi
2. github
3. pythonhosted

so as the alias it is not present on pypi, I have used github, if github would not be available for any reason only the use pythonhosted

File: recipe/conda_build_config.yaml [hunk 2]
```diff
@@ -1,2 +1,2 @@
-MACOSX_SDK_VERSION:            # [osx and x86_64]
-  - 10.15                      # [osx and x86_64]
+#MACOSX_SDK_VERSION:            # [osx and x86_64]
+#  - 10.15                      # [osx and x86_64]
```
Comment (skupr-anaconda) [inline recipe/conda_build_config.yaml]: MACOSX_SDK 10.15 is available already

**Review by skupr-anaconda — COMMENTED** [review]
An error on `aarch64`:
`ps: command not found`

https://github.com/AnacondaRecipes/uwsgi-feedstock/blob/c1f19ba7403162bcabd40fa91e92a6c23b0efafb/recipe/run_test.sh#LL22C16-L22C18

Looks like on that platform is missing  a process viewing command tool `ps`

**Review by skupr-anaconda — APPROVED** [review]
LGTM

**Review by akabanovs — COMMENTED** [review]
@skupr-anaconda @M-Waszkiewicz-Anaconda  should we pin packages like 

pcre
yaml
jansson

in the host section? 

zlib coulb be:

zlib {{ zlib }} 

to indicate the version is taken from the cbc file.

**Comment by M-Waszkiewicz-Anaconda** [comment]
> @skupr-anaconda @M-Waszkiewicz-Anaconda should we pin packages like
> 
> pcre yaml jansson
> 
> in the host section?
> 
> zlib coulb be:
> 
> zlib {{ zlib }}
> 
> to indicate the version is taken from the cbc file.

done

File: recipe/conda_build_config.yaml (hunk 2 above)
Comment (M-Waszkiewicz-Anaconda) [reply recipe/conda_build_config.yaml]: when I uncomment those build fails on OSX

**Review by akabanovs — APPROVED** [review]
LGTM

File: recipe/conda_build_config.yaml (hunk 2 above)
Comment (katietz) [reply recipe/conda_build_config.yaml]: then remove thos lines completely IMHO

**Review by katietz — APPROVED** [review]
otherwise looks good to me.  thanks

**Comment by M-Waszkiewicz-Anaconda** [comment]
> otherwise looks good to me. thanks

leave empty file, or remove the file altogether ?

**Review by sumit0190 — APPROVED** [review]

**Pull request merged**